software/contrib/conway.py

```diff
@@ -0,0 +1,354 @@
+        set_all_bits(self.next_changed_spaces)
+
+    def check_for_stasis(self):
+        """Check the population changes over time to see if we've reached a state of stasis
+        """
+        # we must have at least MAX_DELTAS generations of data
+        if len(self.population_deltas) < self.MAX_DELTAS:
+            return False
+
+        # if there are no changes or everything is dead, we've reached static stasis
+        if self.num_changes == 0 or self.num_alive == 0:
+            return True
+
+        # if the population is oscillating up and down predicatbly, we've probably reached stasis
+        # check for 2, 3, and 4 step repetitions
+        for pattern_length in range(2, 5):
+            sums = []
+            for i in range(self.MAX_DELTAS // pattern_length):
+                sums.append(sum(self.population_deltas[i*pattern_length:i*pattern_length+pattern_length]))
+
+            # check the standard deviation
+            deviation = stdev(sums)
+            mean = sum(sums)/len(sums)
+            if deviation <= 1 and abs(mean) <= 1:
+                return True
+
+        return False
+
+    def main(self):
+        """The main loop for the program
+
+        Handles setting the CV output, drawing to the OLED, and triggering the simulation
+        """
+        turn_off_all_cvs()
+        self.reset()
+
+        in_stasis = False
+
+        while True:
+            # turn off the stasis gate while we calculate the next generation
+            cv6.voltage(0)
+
+            # turn on the FPS gate when we start calculating
+            cv4.voltage(GATE_VOLTAGE)
+
+            # calculate the next generation
+            self.tick()
+
+            # turn off the FPS gate when we're done calculating but before we draw
+            cv4.voltage(0)
+
+            # show the results on the OLED
+            self.draw()
+
+            # check for stasis consitions
+            self.population_deltas.append(self.num_born - self.num_died)
+            if len(self.population_deltas) > self.MAX_DELTAS:
+                self.population_deltas.pop(0)
+            in_stasis = self.check_for_stasis()
+
+            cv1.voltage(MAX_OUTPUT_VOLTAGE * bitwise_entropy(self.field))
+            cv5.voltage(GATE_VOLTAGE if self.num_born > self.num_died else 0)
+
+            # Make sure we don't divide by zero
+            if self.num_alive > 0:
+                cv2.voltage(MAX_OUTPUT_VOLTAGE * self.num_born / self.num_alive)
+            else:
+                cv2.voltage(0)
+
+
+            # Prevent values greater than 1 & division-by-zero errors
+            hi = max(self.num_died, self.num_born)
+            low = min(self.num_died, self.num_born)
+            if (hi > 0):
+                cv3.voltage(MAX_OUTPUT_VOLTAGE * (low/hi))
+            else:
+                cv3.voltage(0)
+
+            # If we've achieved statis, set CV6 & trigger a reset
+            if in_stasis:
+                cv6.voltage(GATE_VOLTAGE)
```

Comment:
It's long enough that before I added the automatic reset functionality you could self-reset by patching `cv6` into `din` and trigger the reset that way.  So I'd definitely say it's useful, thou definitely as a trigger and not as a gate.

Reply:
So should the signal on cv6 be referred to as a trigger rather than a gate in the documentation? I'm not sure if that would be a source of confusion for users or not.

Reply:
Changed the documentation to describe CV6 as a trigger instead of a gate. The gate duration is based on the amount of time it takes to reset the field, so it's a measurable amount, but is fast enough that using it to gate anything would be pretty hard.